### PR TITLE
add blm badge

### PR DESCRIPTION
**Pull request by ladyada**
Add BLM pendant badge BSP

### CODE REVIEW

File: variants/blm_badge/variant.h [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+#define PIN_LED_13           (13u)
+#define PIN_LED_RXL          (13u)
+#define PIN_LED_TXL          (13u)
+#define PIN_LED              PIN_LED_13
+#define PIN_LED2             PIN_LED_RXL
+#define PIN_LED3             PIN_LED_TXL
+#define LED_BUILTIN          PIN_LED_13
+
+// Neopixel LED
+#define INTERNAL_NEOPIXEL_PIN   14
```
Comment (hathach) [inline variants/blm_badge/variant.h:89]: maybe just name this as `PIN_NEOPIXEL` & `NEOPIXEL_NUM` like other variant, which make it easier for library/user sketch.

https://github.com/adafruit/ArduinoCore-samd/blob/master/variants/grand_central_m4/variant.h#L91

File: variants/blm_badge/variant.cpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+  // GPIO 15 & 16 (SWCLK & SWDIO)
+  // --------------------------
+  { PORTA, 30, PIO_TIMER, PIN_ATTR_NONE, No_ADC_Channel, NOT_ON_PWM, NOT_ON_TIMER, EXTERNAL_INT_NONE },
+  { PORTA, 31, PIO_TIMER, PIN_ATTR_NONE, No_ADC_Channel, NOT_ON_PWM, NOT_ON_TIMER, EXTERNAL_INT_NONE },
+
+  // GPIO 15 & 16 (SWCLK & SWDIO)
+  // --------------------------
+  { PORTA, 30, PIO_TIMER, PIN_ATTR_NONE, No_ADC_Channel, NOT_ON_PWM, NOT_ON_TIMER, EXTERNAL_INT_NONE },
+  { PORTA, 31, PIO_TIMER, PIN_ATTR_NONE, No_ADC_Channel, NOT_ON_PWM, NOT_ON_TIMER, EXTERNAL_INT_NONE },
```
Comment (hathach) [inline variants/blm_badge/variant.cpp:77]: seem to be typo, duplicate with above line 71,72

**Review by hathach — COMMENTED** [review]
it is mostly typo, except the variant.c with duplication of pin 15,16 causing issue with pin numbering.

File: boards.txt [hunk 3]
```diff
@@ -1320,3 +1320,40 @@ adafruit_hallowing_m4.menu.usbstack.tinyusb.build.flags.usbstack=-DUSE_TINYUSB
...
+adafruit_blm_badge.build.core=arduino
+adafruit_blm_badge.build.extra_flags=-DCRYSTALLESS -DADAFRUIT_BLM_BADGE -D__SAMD21E18A__ -DARM_MATH_CM0PLUS {build.usb_flags}
+adafruit_blm_badge.build.ldscript=linker_scripts/gcc/flash_with_bootloader.ld
+adafruit_blm_badge.build.openocdscript=openocd_scripts/=blm_badge.cfg
+adafruit_blm_badge.build.variant=blm_badge
+adafruit_blm_badge.build.variant_system_lib=
+adafruit_blm_badge.build.vid=0x239A
+adafruit_blm_badge.build.pid=0x80BF
+adafruit_blm_badge.bootloader.tool=openocd
+adafruit_blm_badge.bootloader.file=gemmaM0/bootloader-gemma_m0-v2.0.0-adafruit.5.bin
```
[the remark this answers is not in the document]
Comment (ladyada) [reply boards.txt]: ok added!

File: variants/blm_badge/variant.cpp (hunk 2 above)
Comment (ladyada) [reply variants/blm_badge/variant.cpp]: oops removed!

File: variants/blm_badge/variant.h [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  See the GNU Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#ifndef _VARIANT_ARDUINO_ZERO_
+#define _VARIANT_ARDUINO_ZERO_
```
[the remark this answers is not in the document]
Comment (ladyada) [reply variants/blm_badge/variant.h]: fixed!

File: variants/blm_badge/variant.h (hunk 1 above)
Comment (ladyada) [reply variants/blm_badge/variant.h]: done!

**Comment by ladyada** [comment]
@hathach ok done please re-review!

**Review by hathach — APPROVED** [review]
look greats now

**Pull request merged**